integration_tests/commands/incr_test.go

```diff
@@ -161,4 +161,144 @@ func TestINCR(t *testing.T) {
 			}
 		})
 	}
-}
\ No newline at end of file
+}
+
+func TestINCRBY(t *testing.T) {
+	conn := getLocalConnection()
+	defer conn.Close()
+
+	type SetCommand struct {
+		key string
+		val int64
+	}
+
+	type IncrByCommand struct {
+		key         string
+		decrValue   any
+		expectedVal int64
+		expectedErr string
+	}
+
+	type GetCommand struct {
+		key         string
+		expectedVal int64
+	}
+
+	testCases := []struct {
+		name           string
+		setCommands    []SetCommand
+		incrByCommands []IncrByCommand
+		getCommands    []GetCommand
+	}{
+		{
+			name: "happy flow",
+			setCommands: []SetCommand{
+				{"key", 3},
+			},
+			incrByCommands: []IncrByCommand{
+				{"key", int64(2), 5, utils.EmptyStr},
+				{"key", int64(1), 6, utils.EmptyStr},
+			},
+			getCommands: []GetCommand{
+				{"key", 6},
+			},
+		},
+		{
+			name: "happy flow with negative increment",
+			setCommands: []SetCommand{
+				{"key", 100},
+			},
+			incrByCommands: []IncrByCommand{
+				{"key", int64(-2), 98, utils.EmptyStr},
+				{"key", int64(-10), 88, utils.EmptyStr},
+				{"key", int64(-88), 0, utils.EmptyStr},
+				{"key", int64(-100), -100, utils.EmptyStr},
+			},
+			getCommands: []GetCommand{
+				{"key", -100},
+			},
+		},
+		{
+			name: "happy flow with unset key",
+			setCommands: []SetCommand{
+				{"key", 3},
+			},
+			incrByCommands: []IncrByCommand{
+				{"unsetKey", int64(2), 2, utils.EmptyStr},
+			},
+			getCommands: []GetCommand{
+				{"key", 3},
+				{"unsetKey", 2},
+			},
+		},
+		{
+			name: "edge case with maxInt64",
+			setCommands: []SetCommand{
+				{"key", math.MaxInt64 - 1},
+			},
+			incrByCommands: []IncrByCommand{
+				{"key", int64(1), math.MaxInt64, utils.EmptyStr},
+				{"key", int64(1), 0, "ERR value is out of range"},
+			},
+			getCommands: []GetCommand{
+				{"key", math.MaxInt64},
+			},
+		},
+		{
+			name: "edge case with negative increment",
+			setCommands: []SetCommand{
+				{"key", math.MinInt64 + 1},
+			},
+			incrByCommands: []IncrByCommand{
+				{"key", int64(-1), math.MinInt64, utils.EmptyStr},
+				{"key", int64(-1), 0, "ERR value is out of range"},
+			},
+			getCommands: []GetCommand{
+				{"key", math.MinInt64},
+			},
+		},
+		{
+			name: "edge case with string values",
+			setCommands: []SetCommand{
+				{"key", 1},
+			},
+			incrByCommands: []IncrByCommand{
+				{"stringkey", "abc", 0, "ERR value is not an integer or out of range"},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			FireCommand(conn, "DEL key")
+			FireCommand(conn, "DEL unsetKey")
+			FireCommand(conn, "DEL stringkey")
```

Comment:
Add defer calls before each of them so that the cleanup happens after the test runs

Reply:
@JyotinderSingh , Made changes in this [commit](https://github.com/DiceDB/dice/pull/676/commits/b678e9ef2568b340652d9dc0acdd7bbb44ecaf24). Also clubbed all keys deletion to single query.